zeppelin-web-angular/e2e/models/notebook-keyboard-page.ts

```
@@ -0,0 +1,1056 @@
+  readonly runButton: Locator;
+  readonly paragraphResult: Locator;
+  readonly newParagraphButton: Locator;
+  readonly interpreterSelector: Locator;
+  readonly interpreterDropdown: Locator;
+  readonly autocompletePopup: Locator;
+  readonly autocompleteItems: Locator;
+  readonly paragraphTitle: Locator;
+  readonly editorLines: Locator;
+  readonly cursorLine: Locator;
+  readonly settingsButton: Locator;
+  readonly clearOutputOption: Locator;
+  readonly deleteButton: Locator;
+
+  constructor(page: Page) {
+    super(page);
+    this.codeEditor = page.locator('.monaco-editor .monaco-mouse-cursor-text');
+    this.paragraphContainer = page.locator('zeppelin-notebook-paragraph');
+    this.firstParagraph = this.paragraphContainer.first();
+    this.runButton = page.locator('button[title="Run this paragraph"], button:has-text("Run")');
+    this.paragraphResult = page.locator('[data-testid="paragraph-result"]');
+    this.newParagraphButton = page.locator('button:has-text("Add Paragraph"), .new-paragraph-button');
+    this.interpreterSelector = page.locator('.interpreter-selector');
+    this.interpreterDropdown = page.locator('nz-select[ng-reflect-nz-placeholder="Interpreter"]');
+    this.autocompletePopup = page.locator('.monaco-editor .suggest-widget');
+    this.autocompleteItems = page.locator('.monaco-editor .suggest-widget .monaco-list-row');
+    this.paragraphTitle = page.locator('.paragraph-title');
+    this.editorLines = page.locator('.monaco-editor .view-lines');
+    this.cursorLine = page.locator('.monaco-editor .current-line');
+    this.settingsButton = page.locator('a[nz-dropdown]');
+    this.clearOutputOption = page.locator('li.list-item:has-text("Clear output")');
+    this.deleteButton = page.locator('button:has-text("Delete"), .delete-paragraph-button');
+  }
+
+  async navigateToNotebook(noteId: string): Promise<void> {
+    if (!noteId) {
+      console.error('noteId is undefined or null. Cannot navigate to notebook.');
```

It seems better to throw an error right away when `noteId` is falsy, since that implies an earlier-step issue, so we don't end up silencing it. WDYT?